easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,92 @@
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for XALT, implemented as an easyblock
+@author: Scott McMillan (NVIDIA)
+"""
+import os
+import re
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY
+from easybuild.tools.build_log import EasyBuildError
+
+
+class EB_XALT(ConfigureMake):
+    """Support for building and installing XALT."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'config_py': [None, "XALT site filter file", MANDATORY],
+            'syshost': ['hardcode:cluster', "System name", CUSTOM],
+            'transmission': ['syslog', "", CUSTOM],
+        }
+        return ConfigureMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configuration step for XALT."""
+
+        # Remove the subdir from the install prefix to resolve the
+        # following configure error.  XALT automatically creates a
+        # versioned directory hierarchy.
+        #
+        #    Do not install XALT with 2.7.27 as part of the
+        #    prefix.  It will lead to many many problems with future
+        #    installs of XALT
+        #
+        #    Executables built with the current version of XALT
+        #    will not work with future installs of XALT!!!
+        #
+        #    If you feel that you know better than the developer
+        #    of XALT then you can configure XALT with the configure
+        #    --with-IrefuseToInstallXALTCorrectly=yes and set the prefix
+        #    to include the version
+        self.installdir = os.path.normpath(re.sub(self.install_subdir, '', self.installdir))
+
+        # XALT site filter config file is mandatory
+        if not self.cfg['config_py']:
+            raise EasyBuildError('XALT site filter config must be specified. '
+                                 'Use "--try-amend=config_py=<file>"')
+        self.cfg.update('configopts', '--with-config=%s' % self.cfg['config_py'])
+
+        if self.cfg['syshost']:
+            self.cfg.update('configopts', '--with-syshostConfig=%s' % self.cfg['syshost'])
+
+        if self.cfg['transmission']:
+            self.cfg.update('configopts', '--with-transmission=%s' % self.cfg['transmission'])
+
+        super(EB_XALT, self).configure_step()
+
+        # Reassemble the default install path
+        self.installdir = os.path.join(self.installdir, self.install_subdir)
+
```

Comment:
I think it needs a function to defined environment variables
```
def make_module_extra(self, *args, **kwargs):
    txt = super(EB_XALT, self).make_module_extra(*args, **kwargs)
    txt += self.module_generator.prepend_paths('LD_PRELOAD', 'lib64/libxalt_init.so')
    txt += self.module_generator.prepend_paths('ALT_LINKER', 'bin/%s' % self.cfg['linker'])
    txt += self.module_generator.prepend_paths('COMPILER_PATH', 'bin')
'ALT_LINKER': '%(installdir)s/bin/ld'
    txt += self.module_generator.set_environment('XALT_EXECUTABLE_TRACKING',  '%s' % self.cfg['executable_tracking'])
    txt += self.module_generator.set_environment('XALT_SCALAR_SAMPLING',  '%s' % self.cfg['scalar_sampling'])
    txt += self.module_generator.set_environment('XALT_DIR', '%s' % self.installdir)
    txt += self.module_generator.set_environment('XALT_ETC_DIR', '%s/etc' % self.installdir)
    if get_software_root('cuda'):
        txt += self.module_generator.set_environment('XALT_GPU_TRACKING', 'yes')
    else:
        txt += self.module_generator.set_environment('XALT_GPU_TRACKING', 'no')
    if self.cfg['transmission'] == "curl":
        txt += self.module_generator.set_environment('XALT_LOGGING_URL', '%s' % self.cfg['transmission_url'])
    return txt
```

Reply:
Agreed, we can handle setting of environment variables in the easyblock, and keep the easyconfigs clean...

Reply:
Now setting the environment variables in the EasyBlock.  Please open a new comment if you have feedback on the latest commit.